src/gui/graphsettings.cpp

```diff
@@ -72,13 +79,13 @@ void GraphSettings::init()
     auto cmap = m_cMgr->defaultCMapKey();
     CMapKey n(userPrefs.value("graphSettings/nodeCMap", cmap.first).toString(),
               userPrefs.value("graphSettings/nodeCMapSize", cmap.second).toInt());
-    m_ui->nodesColor->init(m_cMgr, n, m_exp->modelPlugin()->nodeAttrsScope());
-    m_ui->nodesColor->setVisible(!m_exp->modelPlugin()->nodeAttrsScope().empty());
+    m_ui->nodesColor->init(m_cMgr, n,m_nodeAttrsScope);
+    m_ui->nodesColor->setVisible(!m_nodeAttrsScope.empty());
```

Comment:
note that the user might change/edit the model of an existing experiment
in this scenario, the `m_nodeAttrsScope` should also be updated (note that we have a connection to call this function when the experiment is restarted)

so, after L70 you should do:
``` c++
if (m_exp) {
    Q_ASSERT_X(m_exp->modelPlugin(), "GraphSettings",
               "tried to init the graph settings for a null model!");
    m_nodeAttrsScope = m_exp->modelPlugin()->nodeAttrsScope();
    m_edgeAttrsScope = m_exp->modelPlugin()->edgeAttrsScope();
}
```

However, note that for the `GraphDesigner`, you may also want to re-init those things when the user changes the `nodeAttrsScope` or the `edgeAttrsScope`, right?

Reply:
> However, note that for the GraphDesigner, you may also want to re-init those things when the user changes the nodeAttrsScope or the edgeAttrsScope, right?

Currently, when the user changes either of the scopes, a new GraphWidget is created. However, we could also a couple of methods like ``editNodeAttrsScope`` and ``editEdgeAttrsScope``  and then call ``init()`` instead of initializing a new graph every time a user decides to change the attribute scope.